Sources/Sentry/SentryProfiler.mm

```diff
@@ -108,13 +143,124 @@ - (instancetype)init
         SENTRY_LOG_ERROR(@"SentryProfiler must be initialized on the main thread");
         return nil;
     }
-    if (self = [super init]) {
-        _debugImageProvider = [SentryDependencyContainer sharedInstance].debugImageProvider;
-        _mainThreadID = ThreadHandle::current()->tid();
+
+    if (!(self = [super init])) {
+        return nil;
     }
+
+    _debugImageProvider = [SentryDependencyContainer sharedInstance].debugImageProvider;
+    _mainThreadID = ThreadHandle::current()->tid();
+    _spansInFlight = [NSMutableArray<SentrySpanId *> array];
+    _transactions = [NSMutableArray<SentryTransaction *> array];
     return self;
 }
 
+#    pragma mark - Public
+
++ (void)startForSpanID:(SentrySpanId *)spanID
```

Comment:
That's a good question. Right now this is only called from `- [SentryTracer initWithTransactionContext:...]`, and I assumed that doesn't only get called for root spans.

In practice, these spanIDs are only used to track timed out profiles, and we can't have a situation where a child span lasts longer than a parent span, as the parent span would get stopped/trimmed to the child span end.

Are you thinking this should be renamed to `startForRootSpanID`? I'm not immediately sure if this would require any changes in the profiler logic.

@philipphofmann does everything I said here check out with you?

Reply:
`SentryTracer  initWithTransactionContext` should only be called for root spans, yes. `rootSpan` is even a property on `SentryTracer`.